Add support for sdmmc on stm32l4+ devices

shlomow opened the pull request:
This pr adds the dts bindings for stm32l4+ sdmmc controller, and adds dependency on `USE_STM32_HAL_SD_EX` in `Kconfig`.

shlomow committed:
dts: stm32l4: Support sdmmc on stm32l4+

Add device tree bindings for sdmmc controller
on stm32l4+ devices.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>

gmarull commented on subsys/disk/Kconfig:
```diff
@@ -144,6 +144,7 @@ config DISK_ACCESS_STM32_SDMMC
 	bool "STM32 SDMMC driver"
 	depends on HAS_STM32CUBE
 	select USE_STM32_HAL_SD
+	select USE_STM32_HAL_SD_EX
```
if it's L4 specific, 
```
	select USE_STM32_HAL_SD_EX if SOC_SERIES_STM32L4X
```

shlomow replied:
It's l4+ specific and not l4 specific. I think that it is better to select it globally as it is harmless

erwango replied:
@shlomow it's not harmless as `USE_STM32_HAL_SD_EX` doesn't exist on all series

erwango reviewed (changes requested):
Add L4 dependency

shlomow committed:
disk: stm32: Select `USE_STM32_HAL_SD_EX` on stm32l4

This commit prepares supporting sdmmc on
stm32l4+ devices. When trying to compile the
sdmmc driver there is a compilation error
because `HAL_SDEx_DriveTransceiver_1_8V_Callback`
is not implemented. We solve this by compiling
also `sd_ex` in cube as this function is implemented
there as weak.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>

erwango commented on samples/subsys/fs/fat_fs/boards/nucleo_l4r5zi.overlay:
```diff
@@ -0,0 +1,6 @@
+&sdmmc1 {
```
I might have been too quick in approving this.

If I'm correct, there is no SDMMC peripheral on this board. 
This won't work for users unless they add the component, which at least requires some comments in sample REAMDE or directly in this file.
Interest of this modification is that it allows to see that SDMMC driver compiles on L4+ target, but this would require some additional modifications on sample, so this could be run in CI with command:
` twister -b -c -T samples/subsys/fs/fat_fs/ -p nucleo_l4r5zi`
